xocto/ranges.py

```diff
@@ -790,6 +790,21 @@ def intersection(self, other: Range[datetime.date]) -> Optional["FiniteDateRange
         assert base_intersection.boundaries == RangeBoundaries.INCLUSIVE_INCLUSIVE
         return FiniteDateRange(base_intersection.start, base_intersection.end)
 
+    def is_disjoint(self, other: Range[datetime.date]) -> bool:
+        # Adjacent dates should not be considered disjoint, we extend the other
+        # range to allow them to be considered adjacent.
+        other_start = other.start
+        if other._is_right_inclusive:
+            assert other.start
+            other_start -= datetime.timedelta(days=1)
```

Comment:
:cow: this bug is wider than dates - it also affects any `Range` where the values are discrete. For example: integers.

I think it could also potentially affect other boundary conditions than inclusive-inclusive, although I guess we'd have to go through and figure it out.

I wonder if a wider solution would be some kind of `increment` or `resolution` field? Either directly on the `Range` itself, or coming from a mapping `RANGE_TYPE_RESOLUTIONS = {datetime.Date: datetime.timedelta(days=1), int: 1, ...}`.

(I don't want my comment to block this PR. I'm sure date ranges are our biggest problem in Kraken and your fix as it stands would be a very useful win.)

Reply:
I also think this might be a wider problem, and some kind of resolution might be a good idea (although since ranges can be for any orderable type, it might not always work).

I agree that we shouldn't block here though.

Reply:
I did consider the integer problem here, but I think it's a bit different because if you consider float values then there are plenty of values that exist between 2 and 3.